src/components/marketing-location/marketing-location.service.ts

```diff
@@ -0,0 +1,220 @@
+        'marketingLocation.name',
+        'MarketingLocation with this name already exists.'
+      );
+    }
+
+    const secureProps = [
+      {
+        key: 'name',
+        value: input.name,
+        addToAdminSg: true,
+        addToWriterSg: false,
+        addToReaderSg: true,
+        isPublic: false,
+        isOrgPublic: false,
+        label: 'FieldZoneName',
+      },
+    ];
+
+    try {
+      const query = this.db
+        .query()
+        .call(matchRequestingUser, session)
+        .match([
+          node('rootUser', 'User', {
+            id: this.config.rootAdmin.id,
+          }),
+        ])
+        .call(createBaseNode, 'MarketingLocation', secureProps)
+        .return('node.id as id');
+
+      const result = await query.first();
+      if (!result) {
+        throw new ServerException('Failed to create marketing location');
+      }
+
+      this.logger.info(`marketing location created`, { id: result.id });
+
+      return await this.readOne(result.id, session);
+    } catch (err) {
+      this.logger.error('Could not create marketing location for user', {
+        exception: err,
+        userId: session.userId,
+      });
+      throw new ServerException('Could not create marketing location');
+    }
+  }
+
+  async readOne(id: string, session: ISession): Promise<MarketingLocation> {
+    this.logger.info('readOne', { id, userId: session.userId });
+
+    if (!id) {
+      throw new NotFoundException('no id given');
+    }
+
+    if (!session.userId) {
+      session.userId = this.config.anonUser.id;
+    }
+
+    const readMarketingLocation = this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'MarketingLocation', { id })])
+      .call(matchPermList, 'requestingUser')
+      .call(matchPropList, 'permList')
+      .return('propList, permList, node')
+      .asResult<StandardReadResult<DbPropsOfDto<MarketingLocation>>>();
+
+    const result = await readMarketingLocation.first();
+
+    if (!result) {
+      throw new NotFoundException('MarketingLocation.id', 'id');
+    }
+
+    const secured = parseSecuredProperties(
+      result.propList,
+      result.permList,
+      this.securedProperties
+    );
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...secured,
+    };
+  }
+
+  async update(
+    input: UpdateMarketingLocation,
+    session: ISession
+  ): Promise<MarketingLocation> {
+    const marketingLocation = await this.readOne(input.id, session);
+
+    return await this.db.sgUpdateProperties({
+      session,
+      object: marketingLocation,
+      props: ['name'],
+      changes: input,
+      nodevar: 'marketingLocation',
+    });
+  }
+
+  async delete(id: string, session: ISession): Promise<void> {
+    const marketingLocation = await this.readOne(id, session);
+    try {
+      await this.db.deleteNode({
+        session,
+        object: marketingLocation,
+        aclEditProp: 'canDeleteOwnUser',
```

Comment:
deleting is being refactored. this can be left out for now. we also won't use ACL anymore.